src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs

```diff
@@ -230,18 +234,71 @@ internal static bool TryExtractTracestate(string[] tracestateCollection, out str
 
             if (tracestateCollection != null)
             {
-                var result = new StringBuilder();
+                List<string> listMembers = new List<string>();
+                for (int i = 0; i < tracestateCollection.Length; ++i)
+                {
+                    // test_tracestate_multiple_headers_different_keys
+                    listMembers.AddRange(tracestateCollection[i].Split(','));
+                }
 
-                // Iterate in reverse order because when call builder set the elements is added in the
-                // front of the list.
-                for (int i = tracestateCollection.Length - 1; i >= 0; i--)
+                if (listMembers.Count > 32)
                 {
-                    if (string.IsNullOrEmpty(tracestateCollection[i]))
+                    // https://github.com/w3c/trace-context/blob/master/spec/20-http_request_header_format.md#list
+                    // test_tracestate_member_count_limit
+                    return false;
+                }
+
+                var keySet = new HashSet<string>();
```

Comment:
Should we ignore case when we compare?

Reply:
No, it's not necessary, because the key has been validated to never be upper case at [if (!ValidateKey(key))](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1668/files#diff-ea87cd9e24fd2b50aaf8c211566153790ae78da88bf1ef02d8a97e90557050f3R276) a few lines ahead. (according to [Trace Context v1](https://www.w3.org/TR/trace-context-1/#key) specs)

Reply:
I just added a comment for better readability and reduce confusions for future readers as well.